arch/arm/soc/nxp_kinetis/kwx/soc_kw2xd.c

```diff
@@ -0,0 +1,191 @@
+
+#define ER32KSEL_OSC32KCLK	(0)
+#define ER32KSEL_RTC		(2)
+#define ER32KSEL_LPO1KHZ	(3)
+
+#define TIMESRC_OSCERCLK        (2)
+
+/*
+ * KW2xD Flash configuration fields
+ * These 16 bytes, which must be loaded to address 0x400, include default
+ * protection and security settings.
+ * They are loaded at reset to various Flash Memory module (FTFE) registers.
+ *
+ * The structure is:
+ * -Backdoor Comparison Key for unsecuring the MCU - 8 bytes
+ * -Program flash protection bytes, 4 bytes, written to FPROT0-3
+ * -Flash security byte, 1 byte, written to FSEC
+ * -Flash nonvolatile option byte, 1 byte, written to FOPT
+ * -Reserved, 1 byte, (Data flash protection byte for FlexNVM)
+ * -Reserved, 1 byte, (EEPROM protection byte for FlexNVM)
+ *
+ */
+uint8_t __kinetis_flash_config_section __kinetis_flash_config[] = {
+	/* Backdoor Comparison Key (unused) */
+	0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF,
+	/* Program flash protection; 1 bit/region - 0=protected, 1=unprotected
+	 */
+	0xFF, 0xFF, 0xFF, 0xFF,
+	/*
+	 * Flash security: Backdoor key disabled, Mass erase enabled,
+	 *                 Factory access enabled, MCU is unsecure
+	 */
+	0xFE,
+	/* Flash nonvolatile option: NMI enabled, EzPort enabled, Normal boot */
+	0xFF,
+	/* Reserved for FlexNVM feature (unsupported by this MCU) */
+	0xFF, 0xFF};
+
+static const osc_config_t oscConfig = {
+	.freq = CONFIG_OSC_XTAL0_FREQ,
+	.capLoad = 0,
+
+#if defined(CONFIG_OSC_EXTERNAL)
+	.workMode = kOSC_ModeExt,
+#elif defined(CONFIG_OSC_LOW_POWER)
+	.workMode = kOSC_ModeOscLowPower,
+#elif defined(CONFIG_OSC_HIGH_GAIN)
+	.workMode = kOSC_ModeOscHighGain,
+#else
+#error "An oscillator mode must be defined"
+#endif
+
+	.oscerConfig = {
+		.enableMode = kOSC_ErClkEnable | kOSC_ErClkEnableInStop,
+#if (defined(FSL_FEATURE_OSC_HAS_EXT_REF_CLOCK_DIVIDER) && \
+	FSL_FEATURE_OSC_HAS_EXT_REF_CLOCK_DIVIDER)
+		.erclkDiv = 0U,
+#endif
+	},
+};
+
+static const mcg_pll_config_t pll0Config = {
+	.enableMode = 0U,
+	.prdiv = CONFIG_MCG_PRDIV0,
+	.vdiv = CONFIG_MCG_VDIV0,
+};
+
+static const sim_clock_config_t simConfig = {
+	.pllFllSel = PLLFLLSEL_MCGPLLCLK, /* PLLFLLSEL select PLL. */
+	.er32kSrc = ER32KSEL_RTC,         /* ERCLK32K selection, use RTC. */
+	.clkdiv1 = SIM_CLKDIV1_OUTDIV1(CONFIG_KW2XD_CORE_CLOCK_DIVIDER - 1) |
+		   SIM_CLKDIV1_OUTDIV2(CONFIG_KW2XD_BUS_CLOCK_DIVIDER - 1) |
+		   SIM_CLKDIV1_OUTDIV4(CONFIG_KW2XD_FLASH_CLOCK_DIVIDER - 1),
+};
+
+/** Start modem clock in order to derive MCGOUTCLK */
+static void set_modem_clock(void)
+{
+	/* Ungate PORTB and PORTC clock */
+	SIM->SCGC5 |= SIM_SCGC5_PORTB_MASK | SIM_SCGC5_PORTC_MASK;
+	/* Set PORTB.19 as output - modem RESET pin */
+	GPIOB->PDDR |= 0x00080000u;
+	/* Set PORTC.0 as output - modem GPIO5 pin */
+	GPIOC->PDDR |= 0x00000001u;
+	/* PORTB.19 as GPIO */
+	PORTB->PCR[19] = (PORTB->PCR[19] & ~PORT_PCR_MUX_MASK) |
+			 PORT_PCR_MUX(0x01u);
+	/* PORTC.0 as GPIO */
+	PORTC->PCR[0] = (PORTC->PCR[0] & ~PORT_PCR_MUX_MASK) |
```

Comment:
It is not really board specific, if CLK_OUT from the transceiver is the clock source for the PLL, it will look the same for all boards (these pins are internally connected, not on the board)